### PR TITLE
[Feature] Product Events

### PR DESCRIPTION
**Pull request by MichaelCampbell**
Created CartLogging functionality to encapsulate adding and removing
from the cart.

Created `logAddToCart` function to handle logging of product additions
in the cart.

Created `logRemoveFromCart` function to handle logging of product
removals from the cart.

Updated CartLogging to print out correct price formatting for mParticle

### CODE REVIEW

File: Simcoe/mParticle/MPProductKeys.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ - sku:        The sku of the product.
+ - position:   The position of the product on the screen or impression list.
+ */
+internal enum MPProductKeys: String {
+
+    case brand = "brand"
+    case category = "category"
+    case couponCode = "couponCode"
+    case sku = "sku"
+    case position = "position"
```
Comment (htinlinn) [inline Simcoe/mParticle/MPProductKeys.swift]: We should make these CamelCase, at least until the Swift 3 version.

File: Simcoe/Simcoe.swift [hunk 2]
```diff
@@ -70,96 +70,183 @@ public final class Simcoe {
...
+     - returns: A tracking result.
      */
-    public static func trackLocation(location: CLLocation, withAdditionalProperties properties: Properties?) {
-        engine.trackLocation(location, withAdditionalProperties: properties)
+    public static func logRemoveFromCart(productName: String,
+                                         productId: String,
+                                         quantity: Int,
+                                         price: Double?,
+                                         withAdditionalProperties properties: Properties?) {
+        engine.logAddToCart(productName,
```
Comment (yoyoyoseob) [inline Simcoe/Simcoe.swift]: Shouldn't this be logRemoveFromCart?

Comment (MichaelCampbell) [reply Simcoe/Simcoe.swift]: absolutely! good catch
- [x] Update to use `logRemoveFromCart`

File: Simcoe/mParticle/MPProductKeys.swift (hunk 1 above)
Comment (MichaelCampbell) [reply Simcoe/mParticle/MPProductKeys.swift]: - [x] CamelCase

**Comment by MichaelCampbell** [comment]
@yoyoyoseob @htinlinn please take another look

**Review by yoyoyoseob — APPROVED** [review]

**Pull request merged**